### PR TITLE
fix(ui): use dressing.nvim for better input support

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,24 +25,18 @@ Install `avante.nvim` using [lazy.nvim](https://github.com/folke/lazy.nvim):
...
-      'MeanderingProgrammer/render-markdown.nvim',
-      opts = {
-        file_types = { "markdown", "Avante" },
-      },
-      ft = { "markdown", "Avante" },
-    },
+    "grapp-dev/nui-components.nvim",
+    "MunifTanjim/nui.nvim"
+    --- The below is optional, make sure to setup it properly if you have lazy=true
+    'MeanderingProgrammer/render-markdown.nvim',
```
Comment (yetone) [inline README.md:39]: We can mark render-markdown.nvim as optional, but should we keep its related opts? This way, users can experience the best usage by directly copying the installation code, which is the fastest method.

File: README.md [hunk 2]
```diff
@@ -25,17 +25,21 @@ Install `avante.nvim` using [lazy.nvim](https://github.com/folke/lazy.nvim):
...
+  opts = {
+    -- add any opts here
+  },
   dependencies = {
     "nvim-tree/nvim-web-devicons",
+    "stevearc/dressing.nvim",
+    "nvim-lua/plenary.nvim",
+    --- The below is optional, make sure to setup it properly if you have lazy=true
     {
       "grapp-dev/nui-components.nvim",
```
Comment (yetone) [inline README.md:38]: nui is not optional, right?

Comment (aarnphm) [reply README.md:38]: oh yes my bad let me fix that

File: lua/avante/ui/dressing.lua [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    if vim.bo[bufnr].filetype == C.filetype then
+      state.input_winid = winid
+      state.input_bufnr = bufnr
+      vim.wo[winid].conceallevel = 2
+      vim.wo[winid].concealcursor = "nvi"
+      break
+    end
+  end
+
+  P(state)
```
Comment (aarnphm) [inline lua/avante/ui/dressing.lua:36]:
```suggestion
```

File: lua/avante/ui/dressing.lua [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      state.input_winid = winid
+      state.input_bufnr = bufnr
+      vim.wo[winid].conceallevel = 2
+      vim.wo[winid].concealcursor = "nvi"
+      break
+    end
+  end
+
+
+  local prompt_length = api.nvim_strwidth(fn.prompt_getprompt(state.input_bufnr))
```
Comment (aarnphm) [inline lua/avante/ui/dressing.lua:37]:
```suggestion
  local prompt_length = api.nvim_strwidth(fn.prompt_getprompt(state.input_bufnr))
```

**Pull request merged**